packages/@coorpacademy-components/src/atom/button-link/types.ts

```diff
@@ -25,7 +25,12 @@ const propTypes = {
 
 export type IconType = {
   position: 'right' | 'left';
-  type: keyof typeof ICONS;
+  type?: keyof typeof ICONS;
+  faIconName?: string;
+  faIconColor?: string;
+  faIconBackgroundColor?: string;
+  faIconSize?: number;
```

Comment:
this should help you to avoid using "any" in Button-link component
```suggestion
faIconProps?: PropTypes.InferProps<typeof FaIcon.propTypes>;
```

Reply:
:memo: `faIcon`

Reply:
I got this error when tried the above code:

<img width="978" alt="image" src="https://github.com/CoorpAcademy/components/assets/160047673/28861c4e-8956-4348-9826-585f4fac82f5">

I think the reason is that Icon is wrap with React.Memo, so it only have type of React.Memo return value `React.NamedExoticComponent<object>` which typescript is unable to know the inner props.
This is the same reason that I have to use any to parsed the Icon component to type any to passin props without error (I've updated the code to use other method to pass in the prop without using the "any" type)